server/dns-service/dnsservice-library/src/test/java/org/apache/james/dnsservice/library/netmatcher/NetMatcherTest.java

```diff
@@ -18,95 +18,119 @@
  ****************************************************************/
 package org.apache.james.dnsservice.library.netmatcher;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.LOCALHOST_IP_V4_ADDRESSES_DUPLICATE;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.LOCALHOST_IP_V6_ADDRESSES_DUPLICATE;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.LOCALHOST_IP_V4_ADDRESSES;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.LOCALHOST_IP_V6_ADDRESSES;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.DNS_SERVER_IPV4_MOCK;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.DNS_SERVER_IPV6_MOCK;
+import static org.apache.james.dnsservice.api.mock.DNSFixture.HOST_CANNOT_LOOKUP_BY_NAME;
+
+import org.apache.james.dnsservice.library.inetnetwork.InetNetworkBuilder;
+
 import java.net.UnknownHostException;
-import org.apache.james.dnsservice.api.mock.DNSFixture;
-import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
 import org.junit.Test;
 
-/**
- * Test the NetMatcher class with various IPv4 and IPv6 parameters.
- * 
- */
 public class NetMatcherTest {
 
-    private static NetMatcher netMatcher;
+    private static final String IP_V4_1 = "172.16.0.0/255.255.0.0";
+    private static final String IP_V4_2 = "192.168.1.0/255.255.255.0";
+    private static final String IP_V6_1 = "0:0:0:0:0:0:0:1/32768";
+    private static final String IP_V6_2 = "2781:db8:1234:0:0:0:0:0/48";
+    private static final String IP_V4_IPs = IP_V4_1 + ", " + IP_V4_2;
+    private static final InetNetworkBuilder inetNetworkV4Builder = new InetNetworkBuilder(DNS_SERVER_IPV4_MOCK);
+    private static final InetNetworkBuilder inetNetworkV6Builder = new InetNetworkBuilder(DNS_SERVER_IPV6_MOCK);
+
+    private NetMatcher netMatcher;
 
-    /**
-     * Test for IPV4 uniqueness.
-     */
     @Test
-    public void testIpV4NetworksUniqueness() {
-        netMatcher = new NetMatcher(DNSFixture.LOCALHOST_IP_V4_ADDRESSES_DUPLICATE, DNSFixture.DNS_SERVER_IPV4_MOCK);
-        assertEquals("[172.16.0.0/255.255.0.0, 192.168.1.0/255.255.255.0]", netMatcher.toString());
+    public void matchInetNetworkShouldNotMatchWhenUnknownHostException() throws Exception {
+        netMatcher = new NetMatcher(LOCALHOST_IP_V4_ADDRESSES, DNS_SERVER_IPV4_MOCK);
+
+        assertThat(netMatcher.matchInetNetwork(HOST_CANNOT_LOOKUP_BY_NAME)).isFalse();
     }
 
-    /**
-     * Test for IPV6 uniqueness.
-     */
     @Test
-    public void testIpV6NetworksUniqueness() {
-        netMatcher = new NetMatcher(DNSFixture.LOCALHOST_IP_V6_ADDRESSES_DUPLICATE, DNSFixture.DNS_SERVER_IPV6_MOCK);
-        assertEquals("[0:0:0:0:0:0:0:1/32768, 2781:db8:1234:0:0:0:0:0/48]", netMatcher.toString());
+    public void matchInetNetworkShouldMatchWhenIpV4() throws UnknownHostException {
```

Comment:
I don't understand the test, it requires to jump to other files (in particular because of DNS_SERVER_IPV4_MOCK and test naming)